services/credential-schema/src/schema/schema.service.ts

```
@@ -362,13 +372,32 @@ export class SchemaService {
     // create the new schema with the new version
     let newSchema;
     try {
+      let newStatus: SchemaStatus;
+      if (data.status) {
+        switch (data.status.toUpperCase().trim()) {
+          case 'PUBLISHED':
+            newStatus = SchemaStatus.PUBLISHED;
+            break;
+          case 'DEPRECATED':
+            newStatus = SchemaStatus.DEPRECATED;
+            break;
+          case 'REVOKED':
+            newStatus = SchemaStatus.REVOKED;
+            break;
+          default:
+            newStatus = SchemaStatus.DRAFT;
+            Logger.warn('Unknown schema status, setting to draft');
```

@tushar5526 let me know if we should throw an error here

---

I will add that.

---

have added that as well